src/threading.ts

```diff
@@ -117,12 +118,24 @@ class Thread {
     public writeTraceback(): void {
         this.traceback = "Traceback (most recent call last):\n" + this.traceback;
         process.stdout.write(this.traceback);
+        this.exit();
     }
 
-    // Output the traceback when thread cannot handle an exception, and exit the thread
-    public throwException(): void {
-        this.writeTraceback();
-        this.exit();
+    public throwException(exc: IPy_Object): void {
+        // Whenever an exception occurs, tries to find a handler and if it can't outputs the traceback 
+        this.exc = exc;
+        var f: IPy_FrameObj = this.getTopOfStack();
+        if (!f.tryCatchException(this)) {
+            // Find in previous frames
+            for (var i = this.stack.length - 2; i >= 0; i--) {
```

Comment:
Check the disassembly of the above snippet. ```DUP_TOP``` is the first opcode encountered after the exception is raised in ```LOAD_GLOBAL``` opcode. But the frame stack (of ```nested```) is empty, and we can't duplicate and push an undefined value on the frame stack, as the assertion fails.

That is why I am pushing the exception on the previous stack frames, when you cannot find an exception handler in the current frame. The exception also needs to be present on the stack of the frame that has an exception handler, as it is used by the ```COMPARE_OP``` opcode.